CODE-GUIDELINES.md

```diff
@@ -327,386 +110,290 @@ Follow this comment structure:
 
-### BEM
-
-PatternFly follows the [BEM methodology](http://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/). It's a way to decouple CSS from HTML, and modularise class names so they can be extended.
-
-Class name structure follows the `{{pf-block}}__{{element}}--{{modifier}}` structure:
+```html
+<div class="pf-vertical-rythm">
 
-```sass
-.pf-block__element--modifier {}
-```
+  <h1>Main title</h1>
+  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit</p>
+  <p>Excepteur sint occaecat cupidatat non proident, sunt in</p>
 
-Example:
+  <h2>Secondary title</h2>
+  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit</p>
+  <p>Excepteur sint occaecat cupidatat non proident, sunt in</p>
 
-```html
-<div class="pf-panel pf-panel--collapsible">
-  <div class="pf-panel__title">
-    <h1>Heading</h1>
-  </div>
-
-  <div class="pf-panel__content">
-    <p>Lorem ipsum dolor sit amet.</p>
-  </div>
 </div>
 ```
 
-```sass
-.pf-panel {}                      // Block
-.pf-panel--collapsible {}         // Modifier of block
+#### Spaces, margins and paddings
 
-.pf-panel__title {}               // Element
+PatternFly has a set of spacer variables:
 
-.pf-panel__content {}             // Element
-.pf-panel__content--unpadded {}   // Modifier of element
+```sass
+$pf-spacer-xxxs
+$pf-spacer-xxs
+$pf-spacer-xs
+$pf-spacer-sm
+$pf-spacer-md
+$pf-spacer-lg
+$pf-spacer-xl
+$pf-spacer-xxl
+$pf-spacer-xxxl
+$pf-spacer-xxxxl
 ```
 
-**Why it’s better**
-
-- All the selectors have same specificity.
-- Every element is defined via a block.
-- Every modifier is defined via a block or element.
-- Each class name imparts structural info without binding to exact HTML.
-
-
-
-<!-- ============================================================ -->
-
-## Specificity
-
-Always keep selectors as shallow as possible.
+Always them to define `margin` and `padding`
 
 ```sass
-// Bad
-.header .search-box input[type=search] + .button {}
-
-.pf-search-box {
-  .btn {}
+.nav-link {
+  padding-bottom: $pf-spacer-xxxs;
 }
 
-
-// Good
-.pf-search__button {}
+.dropdown-menu {
+  margin-left: $pf-spacer-sm;
+}
 ```
 
-### Nesting
-
-Avoid Sass nesting, but if you must do it follow the [inception rule](http://thesassway.com/beginner/the-inception-rule) and never go more than three layers deep.
-
-Limit nesting to the following use cases:
+You can also use [Bootstrap spacing utilities](http://v4-alpha.getbootstrap.com/utilities/spacing/) with PatternFly sizes:
 
-1. Media queries
-1. Parent selectors
-1. States, pseudo-classes and pseudo-elements
-1. Overwrite Bootstrap
+```html
+<p class="mb-lg">This paragraph has a large bottom margin</p>
+```
 
+##### Comment all magic numbers
 
-For longer style blocks don't nest the modifier code as it reduced the legibility of the code.
+If a situation arises where a value needs entering into the authoring style sheets that isn't already defined in the variables this should serve as a red flag to you.
```

Comment:
Do you think we need to better define magic number here? We don't explicitly define it.